vm/src/trace.rs

```diff
@@ -236,7 +236,7 @@ impl<P: MemoryProof> Iterator for BlockIter<'_, P> {
 
         let s = &self.block.steps[self.index];
         let inst = parse_u32(s.inst).unwrap();
-        let mut w = parse_alt(&self.block.regs, s.inst);
+        let mut w = parse_alt(&self.regs, s.inst);
```

Comment:
Why this change?

Reply:
self.block.regs is the initial register file for the folding block. It's immutable for the whole k-step execution. Here we need the register file that changes instruction-to-instruction.

Reply:
Moreover, without the change, the smoke tests fail.